extensions/undertow/runtime/src/main/java/io/quarkus/undertow/runtime/UndertowDeploymentRecorder.java

```diff
@@ -378,6 +379,7 @@ public void handle(RoutingContext event) {
                 event.remove(QuarkusHttpUser.AUTH_FAILURE_HANDLER);
                 VertxHttpExchange exchange = new VertxHttpExchange(event.request(), allocator, executorService, event,
                         event.getBody());
+                exchange.setPushHandler(VertxHttpRecorder.getRootHandler());
```

Comment:
Just to be sure: at this point we are entirely sure that the root handler is set in `VertxHttpRecorder`?

Reply:
Yes. At this point the server is processing requests, if the root handler was not set then no request processing would be happening.